src/Stratis.Bitcoin.Features.BlockStore.Tests/BlockStoreTests.cs

```diff
@@ -0,0 +1,241 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Base;
+using Stratis.Bitcoin.Utilities;
+using Xunit;
+using Moq;
+using NBitcoin.DataEncoders;
+
+namespace Stratis.Bitcoin.Features.BlockStore.Tests
+{
+    public class BlockStoreTests
+    {
+        private BlockStore blockStore;
+        private BlockRepositoryMock repository;
+        private ConcurrentChain chain;
+        private IChainState chainState;
+        private NodeLifetime nodeLifetime;
+
+        private ChainedHeader consensusTip;
+
+        public BlockStoreTests()
+        {
+            this.repository = new BlockRepositoryMock();
+
+            this.chainState = new Mock<IChainState>().SetupProperty(x => x.ConsensusTip, this.consensusTip).Object;
+
+            this.chain = this.CreateChain(10);
+            this.consensusTip = this.chain.Tip;
+            this.nodeLifetime = new NodeLifetime();
+
+            this.blockStore = new BlockStore(this.repository, this.chain, this.chainState, new StoreSettings(), this.nodeLifetime, new LoggerFactory());
+        }
+
+        private ConcurrentChain CreateChain(int blocksCount)
+        {
+            var chain = new ConcurrentChain(Network.StratisMain);
+
+            for (int i = 0; i < blocksCount; i++)
+            {
+                var header = new BlockHeader()
+                {
+                    Nonce = RandomUtils.GetUInt32(),
+                    HashPrevBlock = chain.Tip.HashBlock,
+                    Bits = Target.Difficulty1
+                };
+                
+                var chainedHeader = new ChainedHeader(header, header.GetHash(), chain.Tip);
+
+                chain.SetTip(chainedHeader);
+            }
+
+            return chain;
+        }
+
+        private async Task WaitUntilQueueIsEmpty()
+        {
+            while (this.blockStore.BlocksQueueCount != 0)
+                await Task.Delay(100).ConfigureAwait(false);
+            
+            // For very slow environments.
+            await Task.Delay(100).ConfigureAwait(false);
+        }
+
+        /// <summary>Checks that block store tip initializes from the hash of the last saved block.</summary>
+        [Fact]
+        public async Task BlockStoreInitializesTipAtHashBlockAsync()
+        {
+            ChainedHeader initializationHeader = this.chain.Tip.Previous.Previous.Previous;
+            this.repository.BlockHash = initializationHeader.HashBlock;
+
+            await this.blockStore.InitializeAsync().ConfigureAwait(false);
+            Assert.Equal(initializationHeader, this.blockStore.StoreTip);
+        }
+        
+        [Fact]
+        public async Task BlockStoreSavesBatchRightAwayWhenAtConsensusTipAsync()
```

Comment:
It's quite difficult to see which parts of this test (and many of the tests in this class) are the Arrange, Act and Assert.

The testing guidelines say that the test names should follow {MethodName}{GivenContext}{ExpectedOutcome} - but that doesn't appear to have been applied within these tests. 

The guidelines also say 
```
A test must have the following structure(see example below):
Setup test context
NewLine
Call method under test
NewLine
Assert Result
```

In the absense of a testing framework, I suppose having that structure would help identify the Setup, Call, Assert (aka Arrange Act Assert)

Reply:
Fixed